src/app.md

```diff
@@ -725,6 +726,47 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given an array of unparsed version `&str`s, iterate and filter against a semver `VersionReq` and find the latest `Version`.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::Version;
+use semver::VersionReq;
+use std::str::FromStr;
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::SemVerError);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let mut ver = Version::from_str("0.0.0").unwrap();
+    let vset = ["0.9.0", "1.0.0", "1.0.1"];
+    let r = VersionReq::parse("<= 1.0.0").unwrap();
```

Comment:
Let's use `error_chain` and replace `unwrap()` by `?`. If the reader copy paste the snippet and modify the version with a wrong value, the program will exit with a message error more useful than a panic.